pkg/controller/job/job_controller.go

```diff
@@ -785,14 +785,18 @@ func (jm *Controller) manageJob(activePods []*v1.Pod, succeeded int32, job *batc
 						}
 					}
 					if err != nil {
-						defer utilruntime.HandleError(err)
 						// Decrement the expected number of creates because the informer won't observe this pod
-						klog.V(2).Infof("Failed creation, decrementing expectations for job %q/%q", job.Namespace, job.Name)
+						if !apierrors.IsNotFound(err) {
```

Comment:
This is unnecessary here, you can't get `NotFound` when creating pods, and that's what this path is doing.